File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dic={}
        keys=[]
        for i in range(len(nums)):
            if(nums[i] not in keys):
                dic[nums[i]]=1
                keys.append(nums[i])
            else:
                dic[nums[i]]+=1
        dic1 = dict(sorted(dic.items(), key=lambda x: x[1], reverse=True))
        
        print(dic1)
        l=[]
        i=0
        key_list = list(dic1.keys())
        while(k>0):
            l.append(key_list[i])
            i+=1
            k-=1
        return l

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int>res;
        map<int,int>map;
        vector<int>count;
        for(int i=0;i<nums.size();i++){
            map[nums[i]]+=1;
        }
        for(auto i:map){
            count.push_back(i.second);
        }
        sort(count.rbegin(),count.rend());
        int j=0;
        for(int i=0;i<k;i++){
            for(auto i:map){
                if(i.second==count[j]){
                    map[i.first]=0;
                    res.push_back(i.first);
                    j++;
                    break;
                }
            }
        }
        return res;
    }
};